### ISSUE TITLE
Upgrade dependencies golang 1.7, argo and k8s api's 1.21

### ISSUE THREAD

**Issue by kevdowney** [keikoproj/addon-manager#93]
**Is this a BUG REPORT or FEATURE REQUEST?**:
FEATURE
**What happened**:
Let's upgrade dependencies:
* Upgrade to Go 1.7+
* Upgrade Kubernetes APIs (apimachinery, api, etc.) to 1.21
* Upgrade Argo to [v3.2.6](https://github.com/argoproj/argo-workflows/releases/tag/v3.2.6)

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Addon Manager version
- Kubernetes version :
```
$ kubectl version -o yaml
```

**Other debugging information (if applicable)**:
- Addon status:
```
$ kubectl describe addon <addon-name>
```
- controller logs:
```
$ kubectl logs <addon-manager-pod>
```

**Comment by ccfishk** [comment]
Did load--tests comparing CPU/MEM consumption between after API version upgrade and v0.4.3 (master branch). Collected data is shared through file

**Comment by kevdowney** [comment]
Closed by #96
